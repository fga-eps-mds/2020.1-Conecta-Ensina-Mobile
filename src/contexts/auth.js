import AsyncStorage from '@react-native-community/async-storage';
import React, {createContext, useState, useEffect} from 'react';
import {Alert} from 'react-native';
import {HOST} from '@env';

export const AuthContext = createContext({});

export default function AuthProvider({children}) {
  const [user, setUser] = useState(null);
  const [typeUser, setTypeUser] = useState(null);
  const [teacher, setTeacher] = useState(null);
  const [student, setStudent] = useState(null);
  const [loading, setLoading] = useState(true);

  const Host = `${HOST}`;

  useEffect(() => {
    async function loadStorage() {
      const storageTypeUser = await AsyncStorage.getItem('Auth_typeUser');
      const storageUser = await AsyncStorage.getItem('Auth_user');

      if (storageTypeUser) {
        if (JSON.parse(storageTypeUser) === 1) {
          setTypeUser('Adm');
          setLoading(false);
        } else if (JSON.parse(storageTypeUser) === 2) {
          setTypeUser('Professor');
          setLoading(false);
        } else {
          setTypeUser('Aluno');
          setLoading(false);
        }
      }
      if (storageUser) {
        const dataUser = JSON.parse(storageUser);
        let usuario = {
          id: dataUser.id,
          firstName: dataUser.firstName,
          lastName: dataUser.lastName,
          email: dataUser.email,
          password: dataUser.password,
          role: dataUser.role,
          cellphone: dataUser.cellphone,
        };
        setUser(usuario);
        setLoading(false);
        if (dataUser.role == 2 || dataUser.role == 3) {
          const storageStudent = await AsyncStorage.getItem('Auth_student');
          if (storageStudent) {
            const dataStudent = JSON.parse(storageStudent);
            let estudante = {
              birthdate: dataStudent.birthdate,
              grade: dataStudent.grade,
              institution: dataStudent.school,
              cpf: dataStudent.cpf,
              cep: dataStudent.cep,
              number: dataStudent.number,
              details: dataStudent.details,
              description: dataStudent.description,
              special: dataStudent.special,
            };
            setStudent(estudante);
            setLoading(false);
          }
          if (dataUser.role == 2) {
            const storageTeacher = await AsyncStorage.getItem('Auth_teacher');
            if (storageTeacher) {
              const dataTeacher = JSON.parse(storageTeacher);
              let professor = {
                photo: dataTeacher.photo,
                video: dataTeacher.video,
                graduation_area: dataTeacher.graduation_area,
                degree: dataTeacher.degree,
                bank: dataTeacher.bank,
                agency: dataTeacher.agency,
                account: dataTeacher.account,
              };
              setTeacher(professor);
              setLoading(false);
            }
          }
        }
      }
      setLoading(false);
    }
    loadStorage();
  }, []);

  async function signIn(email, password) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: email,
        password: password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/login`, settings);
    try {
      const data = await fetchResponse1.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Login efetuado com sucesso!') {
          console.log('login efetuado');
          storageTypeUser(data.role);
          renderData(data.id, data.role, password);
          if (data.role === 1) {
            setTypeUser('Adm');
            console.log('Adm');
          } else if (data.role === 2) {
            setTypeUser('Professor');
            console.log('Professor');
          } else {
            setTypeUser('Aluno');
            console.log('aluno');
          }
          setUser(data);
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
  } //close signIn function

  async function registerAluno(values, {setStatus}) {
    var ok = false;
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
      }),
    };
    const fetchResponse = await fetch(`${Host}/api/student/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Estudante criado com sucesso!') {
          ok = true;

          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Aluno');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(3);
    }
  } //close registerAluno fuction

  async function registerProf(values, {setStatus}) {
    var ok = false;
    console.log(
      'firstName: ' +
        values.name +
        '\n' +
        'lastName: ' +
        values.surname +
        '\n' +
        'email: ' +
        values.email +
        '\n' +
        'password: ' +
        values.password +
        '\n' +
        'cellphone: ' +
        values.cellphone +
        '\n' +
        'birthdate: ' +
        values.birthdate +
        '\n' +
        'grade: ' +
        values.grade +
        '\n' +
        'institution: ' +
        values.school +
        '\n' +
        'cpf: ' +
        values.cpf +
        '\n' +
        'cep: ' +
        values.cep +
        '\n' +
        'number: ' +
        values.num +
        '\n' +
        'details: ' +
        values.details +
        '\n' +
        'description: ' +
        values.description +
        '\n' +
        'special: ' +
        values.special +
        '\n' +
        'photo: ' +
        values.photo +
        '\n' +
        'video: ' +
        values.video +
        '\n' +
        'graduation_area: ' +
        values.graduation_area +
        '\n' +
        'degree: ' +
        values.degree +
        '\n' +
        'bank: ' +
        values.bank +
        '\n' +
        'agency: ' +
        values.agency +
        '\n' +
        'account: ' +
        values.account +
        '\n',
    );
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        password: values.password,
        cellphone: values.cellphone,
        birthdate: values.birthdate,
        grade: values.grade,
        institution: values.school,
        cpf: values.cpf,
        cep: values.cep,
        number: values.num,
        details: values.details,
        description: values.description,
        special: values.special,
        photo: values.photo,
        video: values.video,
        graduation_area: values.graduation_area,
        degree: values.degree,
        bank: values.bank,
        agency: values.agency,
        account: values.account,
      }),
    };

    const fetchResponse = await fetch(`${Host}/api/teacher/create`, settings);
    try {
      const data = await fetchResponse.json();
      console.log('Success');
      if (data.message) {
        if (data.message === 'Professor criado com sucesso!') {
          ok = true;
          renderData(
            data.data.user.id,
            data.data.user.role,
            data.data.user.password,
          );
          setTypeUser('Professor');
          storageTypeUser(data.data.user.role);
        } else if (data.message.name) {
          if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.email
          ) {
            setStatus({email: 'Email j치 foi registrado'});
          } else if (
            data.message.name === 'SequelizeUniqueConstraintError' &&
            data.message.fields.cpf
          ) {
            setStatus({cpf: 'CPF j치 foi registrado'});
          }
        }
      }
    } catch (error) {
      console.error('Error:', error);
    }
    if (ok) {
      RegistroAlert(2);
    }
  } //close registerProf function

  const RegistroAlert = ({role}) => {
    Alert.alert(
      'Registro',
      'Registro Concluido com sucesso',
      [
        {
          text: 'Finalizar',
          onPress: () => {
            if (role === 2) {
              setTypeUser('Professor');
            } else {
              setTypeUser('Aluno');
            }
          },
        },
      ],
      {cancelable: false},
    );
  };

  async function renderData(id, role, password) {
    var ok = false;
    try {
      let response = await fetch(`${Host}/api/user/` + id);
      const data = await response.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      if (role === 2 || role === 3) {
        ok = true;
      }
      storageUser(usuario);
      setUser(usuario);
    } catch (error) {
      console.error(error);
    }
    if (ok) {
      try {
        let response = await fetch(`${Host}/api/student/` + id);
        const data = await response.json();
        let estudante = {
          birthdate: data.data.student.birthdate,
          grade: data.data.student.grade,
          institution: data.data.student.institution,
          cpf: data.data.student.cpf,
          cep: data.data.student.cep,
          number: data.data.student.number,
          details: data.data.student.details,
          description: data.data.student.description,
          special: data.data.student.special,
        };
        setStudent(estudante);
        storageStudent(estudante);
      } catch (error) {
        console.error(error);
      }
      if (role === 2) {
        try {
          let response = await fetch(`${Host}/api/teacher/` + id);
          const data = await response.json();
          let professor = {
            photo: data.data.teacher.photo,
            video: data.data.teacher.video,
            graduation_area: data.data.teacher.graduation_area,
            degree: data.data.teacher.degree,
            bank: data.data.teacher.bank,
            agency: data.data.teacher.agency,
            account: data.data.teacher.account,
          };
          setTeacher(professor);
          storageTeacher(professor);
        } catch (error) {
          console.error(error);
        }
      }
    }
  }

  async function updateUser(values, id) {
    const settings1 = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        firstName: values.name,
        lastName: values.surname,
        email: values.email,
        cellphone: values.cellphone,
        password: user.password,
      }),
    };
    const fetchResponse1 = await fetch(`${Host}/api/user/${id}`, settings1);

    try {
      const data = await fetchResponse1.json();
      console.log('Success');
    } catch (error) {
      console.error('Error:', error);
    }
    if (user.role === 2 || user.role === 3) {
      const settings2 = {
        method: 'PUT',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          agentRole: user.role,
          birthdate: student.birthdate,
          grade: values.grade,
          institution: values.school,
          cpf: student.cpf,
          cep: values.cep,
          number: values.num,
          details: values.details,
          description: values.description,
          special: student.special,
        }),
      };
      const fetchResponse2 = await fetch(
        `${Host}/api/student/${id}`,
        settings2,
      );
      try {
        const data2 = await fetchResponse2.json();
        console.log('Success');
      } catch (error) {
        console.error('Error:', error);
      }
      if (user.role === 2) {
        const settings3 = {
          method: 'PUT',
          headers: {
            Accept: 'application/json',
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            agentRole: user.role,
            video: values.video,
            graduation_area: teacher.graduation_area,
            degree: values.degree,
            bank: values.bank,
            agency: values.agency,
            account: values.account,
          }),
        };
        const fetchResponse3 = await fetch(
          `${Host}/api/teacher/${id}`,
          settings3,
        );
        try {
          const data3 = await fetchResponse3.json();
          console.log('Success');
        } catch (error) {
          console.error('Error:', error);
        }
      }
    }
    renderData(user.id, user.role, user.password);
  }

  async function storageTypeUser(data) {
    await AsyncStorage.setItem('Auth_typeUser', JSON.stringify(data));
  }

  async function storageUser(data) {
    await AsyncStorage.setItem('Auth_user', JSON.stringify(data));
  }

  async function storageStudent(data) {
    await AsyncStorage.setItem('Auth_student', JSON.stringify(data));
  }

  async function storageTeacher(data) {
    await AsyncStorage.setItem('Auth_teacher', JSON.stringify(data));
  }

  async function signOut() {
    await AsyncStorage.clear().then(() => {
      setUser(null);
      setTypeUser(null);
      setStudent(null);
      setTeacher(null);
    });
  }

  return (
    <AuthContext.Provider
      value={{
        signed: !!user,
        user,
        loading,
        teacher,
        student,
        typeUser,
        Host,
        signIn,
        registerAluno,
        registerProf,
        updateUser,
        signOut,
      }}>
      {children}
    </AuthContext.Provider>
  );
}
