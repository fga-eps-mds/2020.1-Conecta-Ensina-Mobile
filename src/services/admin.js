export async function getTeacherList(Host) {
  try {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);
    const data = await fetchResponse.json();
    if (data.message === 'Nenhum professor pendente') {
      return null;
    } else {
      return data.data.student;
    }
  } catch (error) {
    return error;
  }
}

export async function getReportedUsers(Host) {
  try {
    const fetchResponse = await fetch(`${Host}/api/student/status/2`);
    const data = await fetchResponse.json();
    if (data.message === 'Nenhum professor pendente') {
      return null;
    } else {
      return data.data.student;
    }
  } catch (error) {
    return error;
  }
}

export async function getProfessoUser(Host, id) {
  try {
    console.log(id);
    const response = await fetch(`${Host}/api/student/${id}`);
    const data = await response.json();
    console.log(data);
    return data.data.student;
  } catch (error) {
    return error;
  }
}

export async function statusUpdate(Host, user, id, status) {
  const settings = {
    method: 'PUT',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      status: status,
      agentRole: user.role,
    }),
  };
  try {
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    const data = await fetchResponse1.json();
    return data;
  } catch (error) {
    return error;
  }
}
