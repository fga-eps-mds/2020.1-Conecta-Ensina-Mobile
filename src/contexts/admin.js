import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
