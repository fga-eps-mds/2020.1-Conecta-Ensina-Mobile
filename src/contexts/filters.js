import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
//import * as Subj from '../services/subject';

export const FiltersContext = createContext({});

export default function FiltersProvider({children}) {
  const [filter, setFilter] = useState({
    subject: null,
    dtClass: null,
    duration: null,
    classType: null,
  });
  const {Host} = useContext(AuthContext);
  console.log(filter);

  function setSubjectFilter(subject) {
    let filters = {
      subject: subject,
      dtClass: filter.dtClass,
      duration: filter.duration,
      classType: filter.classType,
    };
    setFilter(filters);
  }

  function setDtClass(date) {
    let filters = {
      subject: filter.subject,
      dtClass: date,
      duration: filter.duration,
      classType: filter.classType,
    };
    setFilter(filters);
  }

  function setDurationFilters(duration) {
    let filters = {
      subject: filter.subject,
      dtClass: filter.dtClass,
      duration: duration,
      classType: filter.classType,
    };
    setFilter(filters);
  }

  function setClassType(type) {
    let filters = {
      subject: filter.subject,
      dtClass: filter.dtClass,
      duration: filter.duration,
      classType: type,
    };
    setFilter(filters);
  }

  return (
    <FiltersContext.Provider
      value={{
        filter,
        setSubjectFilter,
        setDtClass,
        setDurationFilters,
        setClassType,
      }}>
      {children}
    </FiltersContext.Provider>
  );
}
