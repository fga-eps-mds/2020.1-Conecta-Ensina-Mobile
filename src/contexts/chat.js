import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';
import * as Chat from '../services/chat';
import { ClassroomContext } from './classroom';

export const chatContext = createContext({});

export default function ChatProvider({children}) {
  const {Host, user} = useContext(AuthContext);
  const {classroom} = useContext(ClassroomContext);
  const [chat, setChat] = useState(null);

  useEffect(() => {
      readChat()
  }, [])

  async function createChat(text) {
    const response = await Chat.createMessage(
        user.id, 
        text, 
        classroom.id, 
        classroom.teacher, 
        classroom.student, 
        Host)
  }

  async function readChat() {
    console.log(classroom)
    const response = await Chat.getChat(Host, classroom.id);
    if (chat !== response) {
      setChat(response);
    }
  }
  return (
    <chatContext.Provider
      value={{
        readChat,
        createChat,
        chat
      }}>
      {children}
    </chatContext.Provider>
  );
}
