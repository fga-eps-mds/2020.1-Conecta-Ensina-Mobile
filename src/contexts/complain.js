import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Student from '../services/student';

export const ComplainContext = createContext({});

export default function ComplainProvider({children}) {
  const [complain, setComplain] = useState({});
  const {Host, user} = useContext(AuthContext);

  async function createComplain(details,accused) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type':'application/json',
      },
      body:JSON.stringify({
        accused: accused,
        reported_by: user.id,
        details: details,
      })
    }
    const response = await fetch(`${Host}/api/complain/create`, settings)
    try{
      const data = response.json();
      console.log("Complain:", data);
      if (data.message === 'Den√∫ncia criada com sucesso!'){
        return data;
      }
    } catch(error) {
      console.error("error: ", error)
    }
  }

  return (
    <ComplainContext.Provider value={{complain, createComplain}}>
      {children}
    </ComplainContext.Provider>
  );
}