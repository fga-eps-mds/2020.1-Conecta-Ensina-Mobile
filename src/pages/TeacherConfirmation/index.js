import React, {useContext} from 'react';
import Theme from '../../../Theme';
import {AdmContext} from '../../contexts/admin';
import Background1 from '../../components/Background1';
import CustomText from '../../components/CustomText';
import {Container, ContainerGrande, ContainerButton, Button} from './styles';

export default function TeacherConfirmation({route, navigation}) {
  const {item} = route.params;
  const {userDB, teacher, statusUpdate} = useContext(AdmContext);

  const handleSubmit = async () => {
    statusUpdate(item.id);
  };

  return (
    <Theme>
      <Background1>
        <Container>
          <ContainerGrande>
            <CustomText white>{userDB && userDB.firstName}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{userDB && userDB.lastName}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{userDB && userDB.email}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{userDB && userDB.cellphone}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{teacher && teacher.description}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{item.institution}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{teacher && teacher.degree}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{teacher && teacher.graduation_area}</CustomText>
          </ContainerGrande>
          <ContainerGrande>
            <CustomText white>{teacher && teacher.video}</CustomText>
          </ContainerGrande>
        </Container>
        <ContainerButton>
          <Button onPress={handleSubmit}>
            <CustomText white bigSmall>
              Aceitar
            </CustomText>
          </Button>
          <Button onPress={() => navigation.navigate('PendingTeacher')}>
            <CustomText white bigSmall>
              Recusar
            </CustomText>
          </Button>
        </ContainerButton>
      </Background1>
    </Theme>
  );
}
