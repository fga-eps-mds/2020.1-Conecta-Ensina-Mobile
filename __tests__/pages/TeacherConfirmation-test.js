import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {AdmContext} from '../../src/contexts/admin';
import TeacherConfirmation from '../../src/pages/TeacherConfirmation';

describe('Testing Teacher Confirmation', () => {
  const route = {
    params: {
      item: '',
    },
  };
  const statusUpdate = jest.fn();
  const navigation = {
    navigate: jest.fn(),
  };
  const pendingUsers = [
    {
      institution: '',
      description: '',
      degree: '',
      graduation_area: '',
      video: '',
        User: {
          firstName: '',
          lastName: '',
          email: '',
          cellphone: '',
        },
    },
  ];
  const reportedUsers = [
    {
      institution: '',
      description: '',
      degree: '',
      graduation_area: '',
      video: '',
        User: {
          firstName: '',
          lastName: '',
          email: '',
          cellphone: '',
        },
      },
  ];
  const students = {
    institution: '',
    description: '',
    degree: '',
    graduation_area: '',
    video: '',
      User: {
        firstName: '',
        lastName: '',
        email: '',
        cellphone: '',
      },
  };

  test('Should matches snapshot', () => {
    const tree = render(
      <AdmContext.Provider
        value={{students, reportedUsers, pendingUsers, statusUpdate}}>
        <TeacherConfirmation navigation={navigation} route={route} />
      </AdmContext.Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should press Aceitar button', () => {
    const {getByTestId} = render(
      <AdmContext.Provider
        value={{students, reportedUsers, pendingUsers, statusUpdate}}>
        <TeacherConfirmation navigation={navigation} route={route} />
      </AdmContext.Provider>,
    );

    const button = getByTestId('Aceitar');

    fireEvent.press(button);
    //debug();
  });

  test('Should press Recusar button', () => {
    const {getByTestId} = render(
      <AdmContext.Provider
        value={{students, reportedUsers, pendingUsers, statusUpdate}}>
        <TeacherConfirmation navigation={navigation} route={route} />
      </AdmContext.Provider>,
    );

    const button = getByTestId('Recusar');

    fireEvent.press(button);
    //debug();
  });
});
