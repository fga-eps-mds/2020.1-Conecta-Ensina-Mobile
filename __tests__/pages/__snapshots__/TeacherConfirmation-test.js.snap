// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Teacher Confirmation Should matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 10,
        "flexShrink": 1,
        "marginBottom": "5%",
        "opacity": 1,
        "paddingBottom": "10%",
        "paddingLeft": "10%",
        "paddingRight": "10%",
        "paddingTop": "10%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Roboto",
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }
      white={true}
    >
      Selecione as disciplinas em que deseja lecionar
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#184E79",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 4,
            "height": "89%",
            "top": 0,
            "width": "100%",
          },
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#184E79",
                "borderRadius": 9,
                "height": "9%",
                "justifyContent": "center",
                "marginTop": "8%",
                "width": "78%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "display": "flex",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#297FB8",
            "borderRadius": 8,
            "height": 31,
            "justifyContent": "center",
            "marginTop": 30,
            "opacity": 1,
            "width": 138,
          }
        }
        testID="Aceitar"
      >
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Aceitar
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#297FB8",
            "borderRadius": 8,
            "height": 31,
            "justifyContent": "center",
            "marginTop": 30,
            "opacity": 1,
            "width": 138,
          }
        }
        testID="Recusar"
      >
        <Text
          bigSmall={true}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "textAlign": "center",
              },
            ]
          }
          white={true}
        >
          Recusar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
